SU2_DEF/src/SU2_DEF.cpp

```diff
@@ -316,10 +334,105 @@ int main(int argc, char *argv[]) {
+              initialDeformation[iDV] = new su2double[config_container[iZone]->GetnDV_Value(iDV)];
+              for (iDV_Value = 0; iDV_Value < config_container[iZone]->GetnDV_Value(iDV); iDV_Value++) {
+                initialDeformation[iDV][iDV_Value] = config_container[iZone]->GetTotalDeformation(iDV, iDV_Value);
+              }
+            }
+
+            for (convexityCheckIter = 1; convexityCheckIter <= config->GetConvexityCheck_Iter(); convexityCheckIter++) {
+
+              /*--- Recursively change deformation magnitude: decrease if there are nonconvex elements, increase otherwise ---*/
+              deformationDifference /= 2.0;
+
+              if (geometry_container[iZone]->GetnNonconvexElements() > 0) {
+                deformationFactor -= deformationDifference;
+              } else {
+                recursionDepth += 1;
+
+                if (recursionDepth == config_container[iZone]->GetConvexityCheck_Depth()) {
+                  if (rank == MASTER_NODE) {
+                    cout << "Maximum recursion depth reached." << endl;
+                    cout << "Remaining amount of original deformation: " << deformationFactor*100.0 << " percent. " << endl;
+                  }
+                  break;
+                }
+
+                deformationFactor += deformationDifference;              
+              }
+
+              /*--- Load mesh to start every iteration with an undeformed grid 
+              (there must be an easier way to do this with less output, but how?) ---*/
+              CGeometry *geometry_aux = NULL;
+              geometry_aux = new CPhysicalGeometry(config_container[iZone], iZone, nZone);
+              geometry_aux->SetColorGrid_Parallel(config_container[iZone]);
+              geometry_container[iZone] = new CPhysicalGeometry(geometry_aux, config_container[iZone]);
+              delete geometry_aux;
+              geometry_container[iZone]->SetSendReceive(config_container[iZone]);
+              geometry_container[iZone]->SetBoundaries(config_container[iZone]);
+              geometry_container[iZone]->SetPoint_Connectivity();
+              geometry_container[iZone]->SetBoundVolume();
+              if (config_container[iZone]->GetReorientElements()) {
+                geometry_container[iZone]->Check_IntElem_Orientation(config_container[iZone]);
+                geometry_container[iZone]->Check_BoundElem_Orientation(config_container[iZone]);
+              }
+              geometry_container[iZone]->SetEdges(); 
+              geometry_container[iZone]->SetVertex(config_container[iZone]);
+
+              if (config_container[iZone]->GetDesign_Variable(0) != NO_DEFORMATION) {
+                geometry_container[iZone]->SetCoord_CG();
+                geometry_container[iZone]->SetBoundControlVolume(config_container[iZone], UPDATE);
+              }
+              geometry_container[iZone]->PreprocessP2PComms(geometry_container[iZone], config_container[iZone]);
```

Comment:
This should be moved to a function and re-used

Reply:
Done